subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformWithDependenciesIntegrationTest.groovy

```diff
@@ -325,4 +329,220 @@ project(':app') {
+
+        def outputLines = output.readLines()
+        def app1Resolve = outputLines.indexOf("> Task :app:resolveSize")
+        def app2Resolve = outputLines.indexOf("> Task :app2:resolveSize")
+        def lib1Transform = outputLines.indexOf("Single step transform received dependencies files [slf4j-api-1.7.25.jar, common.jar] for processing lib.jar")
+        def lib2Transform = outputLines.indexOf("Single step transform received dependencies files [slf4j-api-1.7.26.jar, common.jar] for processing lib.jar")
+        ![app1Resolve, app2Resolve, lib1Transform, lib2Transform].contains(-1)
+        assert lib1Transform < app1Resolve // scheduled
+        assert lib2Transform > app2Resolve // immediate, TODO wolfs: should be scheduled as well
+    }
+
+    def "transform does not execute when dependencies cannot be found"() {
+        given:
+        mavenHttpRepo.module("unknown", "not-found", "4.3").allowAll().assertNotPublished()
+        buildFile << """
+            project(':app') {
+                task resolve(type: Copy) {
+                    def artifacts = configurations.implementation.incoming.artifactView {
+                        attributes { it.attribute(artifactType, 'size') }
+                    }.artifacts
+                    from artifacts.artifactFiles
+                    into "\${buildDir}/libs"
+                }
+            }        
+            
+            project(':lib') {
+                dependencies {
+                    implementation "unknown:not-found:4.3"
+                }
+            }
+        """
+
+        when:
+        fails "resolve"
+
+        then:
+        output.count('Transforming') == 0
+        failure.assertResolutionFailure(":app:implementation")
+        failure.assertThatCause(Matchers.containsString("Could not find unknown:not-found:4.3"))
+    }
+
+    def "transform does not execute when dependencies cannot be downloaded"() {
+        given:
+        def cantBeDownloaded = mavenHttpRepo.module("test", "cant-be-downloaded", "4.3").publish()
+        cantBeDownloaded.pom.allowGetOrHead()
+        cantBeDownloaded.artifact.expectDownloadBroken()
+
+        buildFile << """
+            project(':app') {
+                task resolve(type: Copy) {
+                    def artifacts = configurations.implementation.incoming.artifactView {
+                        attributes { it.attribute(artifactType, 'size') }
+                    }.artifacts
+                    from artifacts.artifactFiles
+                    into "\${buildDir}/libs"
+                }
+            }        
+            
+            project(':lib') {
+                dependencies {
+                    implementation "test:cant-be-downloaded:4.3"
+                }
+            }
+        """
+
+        when:
+        fails "resolve"
+
+        then:
+        output.count('Transforming') == 4
```

Comment:
Again, the number `4` doesn't really convey what is expected to happen here. I have no better idea than to write out the actual lines we are expecting here.